vm/capi/hash.cpp

```diff
@@ -10,6 +10,14 @@ extern "C" {
     return capi_fast_call(rb_cHash, rb_intern("new"), 0);
   }
 
+  VALUE rb_hash_dup(VALUE self) {
+    return capi_fast_call(self, rb_intern("dup"), 0);
+  }
+
+  VALUE rb_hash_freeze(VALUE self) {
+    return capi_fast_call(self, rb_intern("freeze"), 0);
+  }
```

Comment:
For this you'd use the following files:

* `https://github.com/rubinius/rubinius/blob/master/spec/ruby/optional/capi/hash_spec.rb`: this is used for the actual specs
* `https://github.com/rubinius/rubinius/blob/master/spec/ruby/optional/capi/ext/hash_spec.c`: this is used to expose the C API functions to the above Ruby file
* `https://github.com/rubinius/rubinius/blob/master/spec/ruby/optional/capi/ext/rubyspec.h`: used to "tag" which functions are available